### ISSUE TITLE
sed tool does not accept gzipped format

### ISSUE THREAD

**Issue by AnnaSyme** [galaxyproject/tools-iuc#3860]
Hi all, 

I'm using `Text transformation with sed` on `fastqsanger.gz` files (sed v 1.1.1, on both Galaxy Australia and Galaxy Eu), and it outputs a `fqtoc` file. However, it works ok using an uncompressed file.

Seems to be the same as this issue here: [https://github.com/galaxyproject/galaxy/issues/10176] - if so would it be possible to allow gzipped input into `sed`? 

Example history on Galaxy Australia:
[https://usegalaxy.org.au/u/anna/h/sed-on-compressed-data]

Thank you, Anna

**Comment by wm75** [comment]
> Seems to be the same as this issue here:
>
> galaxyproject/galaxy#10176

Yes, this sounds a lot like the same issue. A "fix" would be just a bit more complicated than in https://github.com/bgruening/galaxytools/pull/1023 and would require a unzip | sed | zip pipeline.

Presumably `sed` is not the only text processing tool that's a candidate for such a "transparent unzipping/rezipping of gzipped text formats" feature. So if that's considered a useful thing to have, we should maybe add it to all relevant tools in https://github.com/bgruening/galaxytools/tree/master/tools/text_processing?

**Comment by bgruening** [comment]
I can only move issues inside an organisation as it seems :(
